src/docfx/docset/Document.cs

```diff
@@ -325,12 +318,12 @@ internal static ContentType GetContentType(string path)
             return ContentType.Page;
         }
 
-        internal static string FilePathToSitePath(string path, ContentType contentType, Schema schema, bool json, bool uglifyUrl)
+        internal static string FilePathToSitePath(string path, ContentType contentType, string mime, bool json, bool uglifyUrl)
         {
             switch (contentType)
             {
                 case ContentType.Page:
-                    if (schema is null || schema.Attribute is PageSchemaAttribute)
+                    if (mime is null || !TemplateEngine.IsData(mime))
```

Comment:
how about having another method `TemplateEngine.IsPage(mime)` which is `!IsData(mime)`. Then the reader would not wonder it's not data, then it's page, implicitly assuming that either `Data` or `Page`.